component/encryptor/encryptor.go

```diff
@@ -0,0 +1,382 @@
+		key = conf.EncryptionKey
+	}
+
+	if key == "" {
+		log.Err("Encryptor::Configure : encryption key not set")
+		return fmt.Errorf("encryption key not set")
+	} else {
+		e.encryptionKey, err = base64.StdEncoding.DecodeString(key)
+		if err != nil {
+			log.Err("Encryptor::Configure : error decoding encryption key")
+			return fmt.Errorf("error decoding encryption key: %w", err)
+		}
+	}
+
+	e.blockSize = 1024 * 1024 // default block size is 1MB
+	if config.IsSet(e.Name() + ".block-size-mb") {
+		e.blockSize = conf.BlockSize * 1024 * 1024
+	}
+
+	e.mountPointCipher = "/mnt/cipher/"
+	if config.IsSet(e.Name() + ".encrypted-mount-path") {
+		e.mountPointCipher = conf.EncryptedMountPath
+	}
+
+	log.Info("Encryptor::Configure : block size %d, encrypted mount path %s", e.blockSize, e.mountPointCipher)
+	return nil
+}
+
+func (e *Encryptor) Start(ctx context.Context) error {
+	return nil
+}
+func (e *Encryptor) CommitData(opt internal.CommitDataOptions) error {
+
+	defer e.handle.Close()
+	lastchunkMeta.Lock()
+	defer lastchunkMeta.Unlock()
+
+	if lastchunkMeta.farthestBlockSeen != -1 {
+		// Write the padding length to the end of the file
+		paddingLengthByte := make([]byte, 8)
+		binary.BigEndian.PutUint64(paddingLengthByte, uint64(lastchunkMeta.paddingLength))
+		endoffset := (lastchunkMeta.farthestBlockSeen + 1) * (int64(e.blockSize) + MetaSize)
+		n, err := e.handle.WriteAt(paddingLengthByte, endoffset)
+		log.Info("Encryptor::CommitData : writing %d bytes, padding length %d at offset %d", n, lastchunkMeta.paddingLength, endoffset)
+		if err != nil {
+			log.Err("Encryptor: Error writing padding length to file: %s", err.Error())
+			return err
+		}
+		lastchunkMeta.farthestBlockSeen = -1
+		lastchunkMeta.paddingLength = 0
+	}
+	return nil
+}
+
+func (e *Encryptor) CreateFile(options internal.CreateFileOptions) (*handlemap.Handle, error) {
+	// Create the file in the mount point
+	log.Info("Encryptor::createFile : %s", e.mountPointCipher+options.Name)
+
+	// Take a map of file handles for parallel read/write to different files ?
+	handle := handlemap.NewHandle(options.Name)
+	if handle == nil {
+		log.Trace("Encryptor::createFile : Failed to create handle for file: %s", options.Name)
+		return nil, syscall.EFAULT
+	}
+	handle.Mtime = time.Now()
+
+	fileHandle, err := os.OpenFile(e.mountPointCipher+options.Name, os.O_RDWR|os.O_CREATE, options.Mode)
+	if err != nil {
+		log.Trace("Encryptor::createFile : Error creating file: %s", err.Error())
+		return nil, err
+	}
+
+	e.handle = fileHandle
+	// Set the file handle in the handle
+	return handle, nil
+}
+
+func formatListDirName(path string) string {
+	// If we check the root directory, make sure we pass "" instead of "/"
+	// If we aren't checking the root directory, then we want to extend the directory name so List returns all children and does not include the path itself.
+	if path == "/" {
+		path = ""
+	} else if path != "" {
+		path = internal.ExtendDirName(path)
+	}
+	return path
+}
```

Comment:
nit: functions that serve only one other function can be made local within the func.
func a() {
    b()
    ...
    func b() { }
}

Reply:
makes sense. But i just picked this method from blobfuse code 
'https://github.com/abhiguptacse/azure-storage-fuse/blob/0352a942523d179489449bd38a8612e84877f44b/component/azstorage/azstorage.go#L223'